format-delimited/src/main/java/io/cdap/plugin/format/delimited/common/SchemaDetector.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.format.delimited.common;
+
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.plugin.format.delimited.dq.DataType;
+import io.cdap.plugin.format.delimited.dq.TypeInference;
+
+import java.util.Comparator;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class that detects schema.
+ */
+public class SchemaDetector {
+
+  private Map<String, EnumSet<DataType>> state = new HashMap<>();
+
+  /**
+   * Adds a data type at the set associated to the column name;
+   *
+   * @param columnName
+   * @param dataType
+   */
+
+  public void addDataType(String columnName, DataType dataType) {
+    if (this.state.containsKey(columnName)) {
+      this.state.get(columnName).add(dataType);
+    } else {
+      this.state.put(columnName, EnumSet.noneOf(DataType.class));
```

Comment:
why are we not adding dataType to the EnumSet here?

Reply:
Seems that I missed it. This is fixed now.